### PR TITLE
Move files to the sweeper and envvar packages

### PR DESCRIPTION
**Pull request by zli82016**
Main changes are in the files 
* mmv1/third_party/terraform/acctest/gcp_sweeper.go
* mmv1/third_party/terraform/sweeper/gcp_sweeper.go
* mmv1/third_party/terraform/acctest/provider_test_utils.go
* mmv1/third_party/terraform/envvar/envvar_utils.go

In this PR, 
1. Move the file `gcp_sweeper.go` from `acctest` package to the `sweeper` package
2. Move most functions in the file `provider_test_utils.go` in the `acctest` package to the package `envvar`
3. Replace the function calls.

The reason to introduce the new packages sweep and envvar is to avoid the import circle.

The `testAccProvider` will be moved to the acctest package from the google package later.

If the file `gcp_sweeper.go` and the env variables are still in the `acctest` package, the sweeper file in the specific service package imports the `acctest` package to add the sweeper to the sweepers map. The `acctest` package imports the `provider` package to initialize the testAccProvider. The `provider` package imports the service package. Then the import circle is formed.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: tpgtools/templates/test_file.go.tmpl [hunk 1]
```diff
@@ -46,6 +46,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 
 	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
```
Comment (roaks3) [inline tpgtools/templates/test_file.go.tmpl:49]: I might just be missing it, but why do we need this import here? I don't see the package being used.

**Review by roaks3 — APPROVED** [review]
Overall LGTM, just one question

File: tpgtools/templates/test_file.go.tmpl (hunk 1 above)
Comment (zli82016) [reply tpgtools/templates/test_file.go.tmpl:49]: Thanks for reviewing the PR.

The package `envvar` is used to create the context map by calling the function `s.ExpandContext`.
https://github.com/GoogleCloudPlatform/magic-modules/blob/2422055fe1c63c36c942b89a029471bba3a263dc/tpgtools/templates/test_file.go.tmpl#L58

`ExpandContext` function uses the `translationMap` which needs the envvar package.
https://github.com/GoogleCloudPlatform/magic-modules/blob/2422055fe1c63c36c942b89a029471bba3a263dc/tpgtools/sample.go#L364

**Comment by zli82016** [comment]
Rebased the main branch to resolve the conflicts.

File: tpgtools/templates/test_file.go.tmpl (hunk 1 above)
Comment (roaks3) [reply tpgtools/templates/test_file.go.tmpl:49]: Interesting, thanks for explaining!

**Pull request merged**